src/extras/helpers/ArrowHelper.js

```diff
@@ -94,6 +92,9 @@ THREE.ArrowHelper.prototype.setLength = function ( length, headLength, headWidth
 	if ( headLength < length ){
 		this.line.scale.set( 1, length - headLength, 1 );
```

Comment:
I am pretty sure it will result in a dot.

BTW, I think your other fixes are excellent. How to best handle the edge case in this particular block of code is a matter of personal preference.